packages/lesswrong/server/callbacks/rateLimits.ts

```diff
@@ -81,6 +234,105 @@ async function enforcePostRateLimit (user: DbUser) {
+  // fetch the comments from within the maxTimeframe
+  const commentsInTimeframe = await getCommentsInTimeframe(user._id, maxHours);
+
+  return await getStrictestCommentRateLimitInfo({
+    commentsInTimeframe, 
+    user, 
+    modRateLimitHours, 
+    modPostSpecificRateLimitHours, 
+    postId
+  });
+}
+
+function getStrictestRateLimitInfo (rateLimits: Array<RateLimitInfo|null>): RateLimitInfo | null {
+  const nonNullRateLimits = rateLimits.filter(rateLimit => rateLimit !== null) as Array<RateLimitInfo>;
+  const sortedRateLimits = nonNullRateLimits.sort((a, b) => {
+    if (a.nextEligible < b.nextEligible) return 1;
+    if (a.nextEligible > b.nextEligible) return -1;
+    return 0;
+  });
+  return sortedRateLimits[0] || null;
+}
+
+function getStrictestPostRateLimitInfo(user: DbUser, postsInTimeframe: Array<DbPost>, modRateLimitHours: number): RateLimitInfo|null {
+  // for each rate limit, get the next date that user could post  
+  const modRateLimitInfo = getModRateLimitInfo(postsInTimeframe, modRateLimitHours, 1)
+  const autoRatelimits = forumSelect(autoRateLimits).posts
+
+  const autoRateLimitInfos = autoRatelimits ? autoRatelimits?.map(
+    rateLimit => getAutoRateLimitInfo(user, rateLimit, postsInTimeframe)
+  ) : []
+
+  const rateLimitInfos = [modRateLimitInfo, ...autoRateLimitInfos]
+
+  return getStrictestRateLimitInfo(rateLimitInfos)
+}
+
+interface StrictestCommentRateLimitInfoParams {
+  commentsInTimeframe: Array<DbComment>,
+  user: DbUser,
+  modRateLimitHours: number,
+  modPostSpecificRateLimitHours: number,
+  postId: string | null
+}
+
+async function getModPostSpecificRateLimitInfo (userId: string, comments: Array<DbComment>, modPostSpecificRateLimitHours: number, postId: string | null): Promise<RateLimitInfo|null> {
+  const eligibleForCommentOnSpecificPostRateLimit = modPostSpecificRateLimitHours > 0 && (await shouldApplyModRateLimitForPost(userId, postId));
+  const commentsOnPost = comments.filter(comment => comment.postId === postId)
+
+  return eligibleForCommentOnSpecificPostRateLimit ? getModRateLimitInfo(commentsOnPost, modPostSpecificRateLimitHours, 3) : null
+}
+
+async function getStrictestCommentRateLimitInfo({commentsInTimeframe, user, modRateLimitHours, modPostSpecificRateLimitHours, postId}: StrictestCommentRateLimitInfoParams): Promise<RateLimitInfo|null> {
+  const commentsOnOthersPostsInInterval =  await getCommentsOnOthersPosts(commentsInTimeframe, user._id)
+  const modGeneralRateLimitInfo = getModRateLimitInfo(commentsOnOthersPostsInInterval, modRateLimitHours, 1)
+
+  const modSpecificPostRateLimitInfo = await getModPostSpecificRateLimitInfo(user._id, commentsOnOthersPostsInInterval, modPostSpecificRateLimitHours, postId)
+
+  const autoRatelimits = forumSelect(autoRateLimits).comments
+  const autoRateLimitInfos = autoRatelimits ? autoRatelimits?.map(
+    rateLimit => getAutoRateLimitInfo(user, rateLimit, commentsInTimeframe)
+  ) : []
```

Comment:
can be simplified to `const autoRateLimitInfos = autoRatelimits?.map(...) ?? []`